packages/fxa-settings/src/components/FormPasswordWithInlineCriteria/index.tsx

```diff
@@ -0,0 +1,382 @@
+  formState: UseFormMethods['formState'];
+  errors: UseFormMethods['errors'];
+  onSubmit: () => void;
+  trigger: UseFormMethods['trigger'];
+  register: UseFormMethods['register'];
+  getValues: UseFormMethods['getValues'];
+  email: string;
+  onFocusMetricsEvent?: () => void;
+  loading: boolean;
+  children?: React.ReactNode;
+  disableButtonUntilValid?: boolean;
+};
+
+const getTemplateValues = (passwordFormType: PasswordFormType) => {
+  let templateValues = {
+    passwordFtlId: '',
+    passwordLabel: '',
+    confirmPasswordFtlId: '',
+    confirmPasswordLabel: '',
+    buttonFtlId: '',
+    buttonText: '',
+  };
+  switch (passwordFormType) {
+    case 'signup':
+      templateValues.passwordFtlId =
+        'form-password-with-inline-criteria-signup-new-password-label';
+      templateValues.passwordLabel = 'Password';
+      templateValues.confirmPasswordFtlId =
+        'form-password-with-inline-criteria-signup-confirm-password-label';
+      templateValues.confirmPasswordLabel = 'Repeat password';
+      templateValues.buttonFtlId =
+        'form-password-with-inline-criteria-signup-submit-button';
+      templateValues.buttonText = 'Create account';
+      break;
+    case 'reset':
+      templateValues.passwordFtlId =
+        'form-password-with-inline-criteria-reset-new-password';
+      templateValues.passwordLabel = 'New password';
+      templateValues.confirmPasswordFtlId =
+        'form-password-with-inline-criteria-confirm-password';
+      templateValues.confirmPasswordLabel = 'Re-enter password';
+      templateValues.buttonFtlId =
+        'form-password-with-inline-criteria-reset-submit-button';
+      templateValues.buttonText = 'Reset password';
+      break;
+  }
+  return templateValues;
+};
+
+export const FormPasswordWithInlineCriteria = ({
+  passwordFormType,
+  formState,
+  errors,
+  onSubmit,
+  email,
+  trigger,
+  register,
+  getValues,
+  onFocusMetricsEvent,
+  loading,
+  children,
+  disableButtonUntilValid = true,
+}: FormPasswordWithInlineCriteriaProps) => {
+  const passwordValidator = new PasswordValidator(email);
+  const [passwordMatchErrorText, setPasswordMatchErrorText] =
+    useState<string>('');
+  const [hasNewPwdFocused, setHasNewPwdFocused] = useState<boolean>(false);
+  const [hasUserTakenAction, setHasUserTakenAction] = useState<boolean>(false);
+  const [isConfirmPwdRequirementsVisible, setIsConfirmPwdCriteriaVisible] =
+    useState<boolean>(false);
+  const [srOnlyPwdFeedbackMessage, setSROnlyPwdFeedbackMessage] =
+    useState<string>();
+  const [srOnlyConfirmPwdFeedbackMessage, setSROnlyConfirmPwdFeedbackMessage] =
+    useState<string>();
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedPasswordMatchError = ftlMsgResolver.getMsg(
+    'form-password-with-inline-criteria-match-error',
+    'Passwords do not match'
+  );
+
+  const templateValues = getTemplateValues(passwordFormType);
+
+  // Timeout to delay showing/hiding info, prevents jankiness
```

Comment:
Is all of this code for delay and showing/hiding password feedback still needed?

Reply:
Nice catch, meant to circle back and remove this.